my_pyscf/mcpdft/laspdft.py

```diff
@@ -227,9 +232,12 @@ def optimize_mcscf_(self, mo_coeff=None, ci0=None, **kwargs):
             Has the same calling signature as the parent kernel method. '''
             with _mcscf_env(self):
                 if self.DoLASSI:
-                    self.e_states = self.e_roots
-                    self.statlis = [x for x in range(len(self.e_states))]
-                    self.fcisolver.nroots = len(self.e_states) if self.states is None else self.states
+                    self.statlis = [x for x in range(len(self.e_roots))] # LASSI-LPDFT
+                    if self.states is None:
+                        self.fcisolver.nroots = len(self.e_roots)
+                        self.states = list(range(len(self.e_roots)))
+                    else:
+                        self.fcisolver.nroots = self.states
```

Comment:
Can self.fcisolver.nroots be set to a list? Did you mean self.fcisolver.nroots = len (self.states)?